mattermost/mattermost-webapp
# [MM-22802] [MM-22801] Dont show createPost until channel roles are loaded

#### Summary
- Currently reloading the page while on a channel results in the create post text box showing the channel as read only for a split second this is because while the roles are not loaded into the state the `haveIChannelPermission` returns false for the `create_post` permission.

- This PR makes it so that the `createPost` component is not rendered (i.e. the loading indicator is shown) if the channel's roles are not in the state.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22801
https://mattermost.atlassian.net/browse/MM-22802

#### Test steps
- Reload page while on a channel
- Ensure that the text box does not flash as read only
- Ensure that the text box is in focus after reloading
- Repeat test with some channel moderation setting enabled
